decentraland/unity-renderer
# feat: gpu skinning tests

Adding a visual test to GPU Skinning. All the animations have been moved to Resources so we can use them in tests.

The test could be greatly simplified without all the nuisances of the loading of a wearable and animate it.

We have to:
1) Load a wearable (ensuring that we are not reusing the material -_-).
2) Add an animation component to the proper GameObject (the root of the wearable hierarchy).
3) Apply the mesh combiner (it could be avoided but it greatly simplifies the issue by having a single SkinnedMeshRenderer)..
4) Freeze the animation in a frame (which is trivial with an Animator but a challenge with Animation)

At this point we are ready to apply the GPU Skinning.

So the test creates two instances of all the above, applies the GPUSkinning to one of them and takes a snapshot.

## How to test the changes?
If the tests passes, all is good. You can manually go to the CI artifacts and check the test screenshot yourself.

## Our Code Review Standards
https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md
